saleor/graphql/payment/mutations.py

```diff
@@ -76,18 +93,29 @@ class Meta:
         error_type_field = "payment_errors"
 
     @classmethod
-    def clean_payment_amount(cls, info, checkout_total, amount):
-        if amount != checkout_total.gross.amount:
+    def clean_payment_amount(cls, info, partial, checkout, checkout_total, amount):
+        if not partial and amount != checkout_total.gross.amount:
             raise ValidationError(
                 {
                     "amount": ValidationError(
-                        "Partial payments are not allowed, amount should be "
-                        "equal checkout's total.",
+                        "Amount does not cover checkout amount and "
+                        "the payment is not marked as partial.",
                         code=PaymentErrorCode.PARTIAL_PAYMENT_NOT_ALLOWED,
                     )
                 }
             )
 
+        remaining = checkout_total.gross - get_covered_balance(checkout)
+        if amount > remaining.amount:
+            raise ValidationError(
+                {
+                    "amount": ValidationError(
+                        "Amount should not exceed checkout's total.",
```

Comment:
Shouldn't be a `checkout's remaining amount`? Or something to emphasize, that exception is raised because provided amount is bigger that remaining amount for checkout